src/StaticWebAssetsSdk/Tasks/WriteImportMapToHtml.cs

```diff
@@ -0,0 +1,101 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Diagnostics.CodeAnalysis;
+using System.Text.RegularExpressions;
+using Microsoft.Build.Framework;
+using Microsoft.Build.Utilities;
+
+namespace Microsoft.AspNetCore.StaticWebAssets.Tasks;
+
+public class WriteImportMapToHtml : Task
+{
+    [Required]
+    public ITaskItem[] Assets { get; set; } = [];
+
+    [Required]
+    public ITaskItem[] Endpoints { get; set; } = [];
+
+    [Required]
+    public string ManifestType { get; set; } = string.Empty;
+
+    [Required]
+    public string Source { get; set; } = string.Empty;
+
+    [Required]
+    public string OutputPath { get; set; } = string.Empty;
+
+    [Required]
+    public ITaskItem[] HtmlFiles { get; set; } = [];
+
+    [Output]
+    public ITaskItem[] HtmlCandidates { get; set; } = [];
+
+    [Output]
+    public ITaskItem[] HtmlFilesToRemove { get; set; } = [];
+
+    [Output]
+    public string[] FileWrites { get; set; } = [];
+
+    [SuppressMessage("Performance", "SYSLIB1045:Convert to 'GeneratedRegexAttribute'.", Justification = "The assembly targets .NET framework as well")]
+    private static readonly Regex _assetsRegex = new Regex(@"@Assets\['([^']*)'\]");
+
+    [SuppressMessage("Performance", "SYSLIB1045:Convert to 'GeneratedRegexAttribute'.", Justification = "The assembly targets .NET framework as well")]
+    private static readonly Regex _importMapRegex = new Regex(@"<!--\s*ImportMap\s*-->");
+
+    public override bool Execute()
+    {
+        var manifest = GenerateStaticWebAssetEndpointsManifest.CreateManifest(Log, Assets, Endpoints, ManifestType, Source);
+        var resources = ResourceCollectionResolver.ResolveResourceCollection(manifest);
+        var importMap = ImportMapDefinition.FromResourceCollection(resources);
```

Comment:
I think we should operate directly at the `Endpoint` level on the ImportMapDefinition and avoid all the stuff with regards to resource collection, etc. It brings a lot of unnecessary complexity and duplication to the SDK for what is essentially finding all the endpoints that are JS modules and writing a JSON object in a specific format.